pycqed/measurement/waveform_control/segment.py

```diff
@@ -754,7 +754,7 @@ def get_segment_start_end(self):
             self.resolve_segment(store_segment_length_timer=False)
         return np.min(start_end_times[:, 0]), np.max(start_end_times[:, 1])
 
-    def _test_overlap(self):
+    def _test_overlap(self, allow_overlap=False, tol=1e-12):
```

Comment:
It seems that you cherry-picked an improvement for a feature that does not exist on the master. allow_overlap is part of MR !3. It won't work on this branch because it also needs changes in other functions.

I anyways do not see the relation of this to the current PR. It rather seems to be related to MR !3, and I can cherry-pick it there.